components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/wsdl/AbstractWSDLProcessor.java

```diff
@@ -156,4 +162,23 @@ else if (scheme.startsWith("https")) {
 
         return transports;
     }
+
+    /**
+     * Creates a zip for from the files of the provided path and returns an InputStream for the created zip file.
+     *
+     * @param wsdlArchiveFilePath file path where the WSDL archive is extracted
+     * @return InputStream object of the WSDL archive zip file
+     * @throws APIMgtWSDLException when error occurred when creating the zip and returning the InputStream
+     */
+    InputStream getWSDLArchive(String wsdlArchiveFilePath) throws APIMgtWSDLException {
+        try {
+            Collection<File> allFiles = FileUtils.listFiles(new File(wsdlArchiveFilePath), null, true);
+            String outputZipFile = wsdlArchiveFilePath + File.separator + APIConstants.WSDL_ARCHIVE_UPDATED_ZIP_FILE;
+            ZIPUtils.zipFiles(outputZipFile, allFiles);
+            return new FileInputStream(outputZipFile);
```

Comment:
On second thoughts, you could do something like this,

try (FileInputStream in =  getWSDLArchive(wsdlArchiveFilePath)) {
     // Use the  stream
}

Reply:
This is actually used for REST APIs. The InputStream will be directly passed to the CXF layer to serve as the Response. So I don't think it can be closed before serving the content. 

I was under the impression that the CXF layer should close this after serving the request as we do not have full control of the InputStream after passing it to the CXF layer. 

Anyway, I will look into a different option rather than using an InputStream as the return type. Most probably using a byte[] from the file rather than using its stream.

Reply:
Ya in this case we can copy the contents of the File to a ByteArrayInputStream and then close the file ourselves internally. Then we can serve the REST API from the ByteArrayInputStream and not have to worry about it being closed. WDYT?

Reply:
Fixed it in that way 👍